### DISCUSSION TITLE
PHP8 null style fixer

### DISCUSSION THREAD

**Discussion by pkly** [PHP-CS-Fixer/PHP-CS-Fixer#6971] [Feature ideas]
From 7.1 onwards we've been able to use `?` to show nullability, but PHP8 introduces better typing, so I was wondering if there are any plans to create a fixer that'd replace something like `?Foo` to `Foo|null`?

**Comment by Wirone** [comment]
Hi @pkly 🙂. There's #3562 proposal, but unfortunately it is still waiting to be picked up by someone. Are in interested 😉?

**Reply by pkly** [reply]
I could probably try it. Any suggested null stuff I could look at as an example?

**Reply by Wirone** [reply]
You may want to look at [`nullable_type_declaration_for_default_null_value`](https://cs.symfony.com/doc/rules/function_notation/nullable_type_declaration_for_default_null_value.html) 🙂.

**Reply by paulbalandan** [reply]
I have created an implementation that works both ways. Would it be rude to send that as PR while there is existing?

**Reply by Wirone** [reply]
@paulbalandan as a maintainers we want the best solution, so having 2 implementations allow us choose better one (in terms of maintainability, complexity etc). Maybe we can somehow merge them into one and make both you and @pkly a co-authors? 🙂

**Reply by pkly** [reply]
I wouldn't mind.

**Comment by Wirone** [comment]
Recently [`nullable_type_declaration`](https://cs.symfony.com/doc/rules/language_construct/nullable_type_declaration.html) was introduced and can standardise nullability both ways (question mark or union type).
